### PR TITLE
core(critical-request-chains): prune requests without an initiator

### PR DESCRIPTION
**Pull request by patrickhulce**
Fixes master CI unit tests

**Summary**
The slimmer third-party-web bundle in #11137 dropped the obscure entity we used for testing so I updated to one that won't disappear.

Hopefully we can reenable required checks again soon and get rid of CI that's always red to reserve the overall X signal 😅 

UPDATE: after this PR was initially written, #11148 was autolanded before the required checks were turned back on which had critical request chain issues.

### CODE REVIEW

**Comment by paulirish** [comment]
restarted these tests but this one is still failing 

image

**Comment by patrickhulce** [comment]
> restarted these tests but this one is still failing

Yikes, so in between the time of fixing this one, @devtools-bot also auto-landed https://github.com/GoogleChrome/lighthouse/pull/11148 before `unit_ubuntu-latest` was required thanks to `land-when-ci-is-green`  🤦 

This is actually a nice prompt to fix a long-standing bug in critical request chains we never ran into during our tests :) 

Before, when a script is to the head added via script and doesn't have an `initiatorRequest` it automatically becomes "critical" because we only look at the priority in that audit and not any broader graph context. This led to weird dangling requests in CRC audit that weren't even part of the root document chain (see the identification of `favicon.ico` in sample json update as the "longest chain" which never made sense).

Now, we'll ignore any request in CRC that didn't have an initiator.

Long-term I think we should refactor critical request chains to be based on the page-dependency-graph instead of manually computing using a set of heuristics to approximate the graph.


This is a bigger change though, so the best course of action might be revert https://github.com/GoogleChrome/lighthouse/pull/11148, land this with just the 3p 1-liner fix and then reland https://github.com/GoogleChrome/lighthouse/pull/11148 with these CRC changes.

**Comment by patrickhulce** [comment]
Oof, actually my suggested course of action would require turning back off `unit_ubuntu-latest` as a required check since a revert of just #11148 won't pass CI 😬 Given how much trouble that got me into here, maybe we do just get this green again with the PR as-is?

File: lighthouse-core/test/computed/critical-request-chains-test.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ describe('CriticalRequestChain gatherer: extractChain function', () => {
     }
   );
 
-  it('returns correct data for two parallel chains', () => {
```
Comment (patrickhulce) [inline lighthouse-core/test/computed/critical-request-chains-test.js:119]: these tests were mostly 3-4 years old when this audit was very different, no real-world trace I found had a situation where this was desirable

File: lighthouse-core/test/computed/critical-request-chains-test.js [hunk 2]
```diff
@@ -168,11 +159,11 @@ describe('CriticalRequestChain gatherer: extractChain function', () => {
     });
   });
 
-  it('returns empty chain list when no critical request', () => {
```
Comment (patrickhulce) [inline lighthouse-core/test/computed/critical-request-chains-test.js:171]: this is an impossible case

File: lighthouse-core/test/computed/critical-request-chains-test.js [hunk 3]
```diff
@@ -182,26 +173,10 @@ describe('CriticalRequestChain gatherer: extractChain function', () => {
     assert.deepEqual(criticalChains, {});
   });
 
-  it('returns two single node chains for two independent requests', () => {
```
Comment (patrickhulce) [inline lighthouse-core/test/computed/critical-request-chains-test.js:185]: same case as the above (pruning non-root document)

File: lighthouse-core/test/computed/critical-request-chains-test.js [hunk 4]
```diff
@@ -222,27 +197,27 @@ describe('CriticalRequestChain gatherer: extractChain function', () => {
               },
             },
           },
-        },
-      },
-      4: {
```
Comment (patrickhulce) [inline lighthouse-core/test/computed/critical-request-chains-test.js:227]: just whitespace indentation

File: lighthouse-core/test/results/sample_v2.json [hunk 5]
```diff
@@ -470,21 +470,12 @@
                 }
               }
             }
-          },
-          "75994.33": {
-            "request": {
-              "url": "http://localhost:10200/favicon.ico",
```
Comment (patrickhulce) [inline lighthouse-core/test/results/sample_v2.json:476]: 🎉

File: lighthouse-core/test/results/sample_v2.json [hunk 6]
```diff
@@ -470,21 +470,12 @@
...
-            }
           }
         },
         "longestChain": {
-          "duration": 5536.498000001302,
-          "length": 1,
-          "transferSize": 221
+          "duration": 4796.288000012282,
+          "length": 2,
+          "transferSize": 30174
```
Comment (patrickhulce) [inline lighthouse-core/test/results/sample_v2.json:478]: huge win IMO that `favicon.ico` at 221 is no longer flagged as the worst chain 🎉

**Comment by patrickhulce** [comment]
just talked to @paulirish in 1:1 and he's still 👍 on this and it's sat for about a day for folks to object soooo I'm gonna merge in interest of green. if there are strong feelings about the CRC audit then happy to bring up in a future refactoring PR :)

**Pull request merged**